packages/bundle-source/test/endo-script-format.test.js

```diff
@@ -4,28 +4,43 @@ import test from '@endo/ses-ava/prepare-endo.js';
 import * as url from 'url';
 import bundleSource from '../src/index.js';
 
-const generate = async (options = {}) => {
-  const entryPath = url.fileURLToPath(
-    new URL(`../demo/meaning.js`, import.meta.url),
-  );
+/**
+ * @template {Partial<object>} Options
+ * @param {string} entry
+ * @param {Options} options
+ */
+const generate = async (entry, options = {}) => {
+  const entryPath = url.fileURLToPath(new URL(entry, import.meta.url));
   return bundleSource(entryPath, {
     format: 'endoScript',
     ...options,
   });
 };
 
 test('endo script format', async t => {
-  const bundle = await generate();
+  const bundle = await generate('../demo/meaning.js');
   t.is(bundle.moduleFormat, 'endoScript');
   const { source } = bundle;
   const compartment = new Compartment();
   const ns = compartment.evaluate(source);
   t.is(ns.meaning, 42);
 });
 
+test('endo script format supports typescript type erasure', async t => {
```

Comment:
I think we need more test cases to help make clear what to expect.

IIUC this these work:
```ts
// tsFromTs.ts
export { fortune } from './fortune.ts
```

```ts
// tsFromJs.ts
export { meaning } from './meaning.js
```

And this won't,
```ts
// jsFromTs.js
export { fortune } from './fortune.ts
```

Reply:
I’ve posted tests alongside these that verify that the full cross-product of {js,ts} can import and export {js,ts}, and also an expected failing test that you can’t `import 'fortune.js'` in TypeScript, if there is no `fortune.js`, even if there _is_ `fortune.ts`.